src/ALE/MOM_ALE.F90

```diff
@@ -974,6 +976,89 @@ subroutine ALE_remap_velocities(CS, G, GV, h_old, h_new, u, v, OBC, dzInterface,
 
 end subroutine ALE_remap_velocities
 
+!> Interpolate to find an updated array of values at interfaces after remapping.
+subroutine ALE_remap_interface_vals(CS, G, GV, h_old, h_new, int_val)
+  type(ALE_CS),                              intent(in)    :: CS       !< ALE control structure
+  type(ocean_grid_type),                     intent(in)    :: G        !< Ocean grid structure
+  type(verticalGrid_type),                   intent(in)    :: GV       !< Ocean vertical grid structure
+  real, dimension(SZI_(G),SZJ_(G),SZK_(GV)), intent(in)    :: h_old    !< Thickness of source grid
+                                                                       !! [H ~> m or kg m-2]
+  real, dimension(SZI_(G),SZJ_(G),SZK_(GV)), intent(in)    :: h_new    !< Thickness of destination grid
+                                                                       !! [H ~> m or kg m-2]
+  real, dimension(SZI_(G),SZJ_(G),SZK_(GV)+1), &
+                                             intent(inout) :: int_val  !< The interface values to interpolate [A]
+
+  real :: val_src(GV%ke)  ! A column of interface values on the source grid [A]
+  real :: val_tgt(GV%ke)  ! A column of interface values on the target grid [A]
+  real :: h_src(GV%ke)    ! A column of source grid layer thicknesses [H ~> m or kg m-2]
+  real :: h_tgt(GV%ke)    ! A column of target grid layer thicknesses [H ~> m or kg m-2]
+  real :: h_neglect, h_neglect_edge  ! Tiny thicknesses used in remapping [H ~> m or kg m-2]
+  logical :: show_call_tree
+  integer :: i, j, k, nz
+
+  do j=G%jsc,G%jec ; do i=G%isc,G%iec ; if (G%mask2dT(i,j)>0.) then
+    do k=1,nz
```

Comment:
nz is never set in this subroutine or the one below. I'm guessing this might be the source of some of my problems with bad data appearing after a time step.

Reply:
It crashes here when compiled in debug mode at least. I think this might get optimized away when optimizations are on, so there may also still be a problem in `remap_OBC_fields`, but I can't debug that right now because of the failure here.

Reply:
Thank you for finding this, @andrew-c-ross.  I had tested this code when I first developed it (in a different file), but then I had to refactor it to harmonize it with some other substantial refactoring, and apparently I was not careful enough when cutting and pasting this code between files.   I will add the lines setting nz in these files, and then do a force-push of the corrections to the last two commits in this PR.

Reply:
I have updated the PR (via a force-push) to correct this bug, and I have tested this using the debug settings to verify that the newly added code does now work as intended.  Thank you again, @andrew-c-ross, for helping to track these problems down.